flake.nix

```diff
@@ -0,0 +1,58 @@
+{
+  description = "Next generation distributed, event-driven, parallel config management!";
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:urandom2/nixpkgs/mgmt";
+  };
+  outputs = {
+    flake-utils,
+    nixpkgs,
+    self,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      buildInputs = with pkgs; [
+        augeas
+        libvirt
+        libxml2
+      ];
+      nativeBuildInputs = with pkgs; [
+        gotools
+        nex
+        pkg-config
+        ragel
+      ];
+      pkgs = nixpkgs.legacyPackages.${system};
+    in {
+      devShells.default = pkgs.mkShell {
+        inherit buildInputs nativeBuildInputs;
+        packages = [pkgs.go];
```

Comment:
if there is anything I missed that is useful for testing/debugging/development, we should add it here